NARFO_BE/NARFO_BE/Controllers/MembersController.cs

```diff
@@ -2,121 +2,133 @@
 
         public MembersController(narfoContext context)
         {
             _context = context;
             if (!_context.Members.Any())
             {
-                Members newMember = new Members();
-                newMember.LastName = "admin";
+                _Member newMember = new _Member();
+                newMember.SURNAME = "admin";
                 newMember.Username = "admin";
-                newMember.FirstName = "admin";
-                newMember.Password = "admin";
+                newMember.Firstname = "admin";
+                newMember.Email = "admin@admin.com";
+               
                 _context.Members.AddAsync(newMember);
                 _context.SaveChanges();
             }
         }
-        
-        public void ListofMembers(List<Members> members)
-        {
-            this.members = members;
-        }
-        
 
-        // GET: api/Members
-        [HttpGet]
-        public async Task<ActionResult<IEnumerable<Members>>> GetMembers()
-        {
-            return await _context.Members.ToListAsync();
-        }
-        public IEnumerable<Members> GetAllMembers()
+        public void ListofMembers(List<_Member> members)
         {
-            return members;
+            this.members = members;
         }
 
-        // GET: api/Members/5
-        [HttpGet("{id}")]
-        public async Task<ActionResult<Members>> GetMembers(int id)
+        //
+        // POST: /Account/Login
+        [HttpPost("get/login")]
+        public async Task<ActionResult <_Member>> Login([FromBody] _Member model)
         {
-            var members = await _context.Members.FindAsync(id);
+            
+            _Member user = await _context.Members.FirstOrDefaultAsync(member=>member.Email == model.Email && member.Password == encryption.ComputeHash(model.Password));
 
-            if (members == null)
+            if (user == null)
             {
-                return NotFound();
+                return BadRequest(new { status = "Failed", message = "Invalid login" });
             }
+            else {
 
-            return members;
+                return Ok(new { status = "Success", member = model });
+            }
+            
+            
         }
 
-        // PUT: api/Members/5
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutMembers(int id, Members members)
+        private ActionResult<_Member> Json(object p)
         {
-            if (id != members.Id)
-            {
-                return BadRequest();
-            }
+            throw new NotImplementedException();
+        }
 
-            _context.Entry(members).State = EntityState.Modified;
 
-            try
+        // GET: Member/Email
+        [HttpGet("get/all/user")]
+        public  async Task<ActionResult<MemberPrototype>> GetMembersEmail()
+        {
+           
+           List<MemberPrototype> endpoint = new List<MemberPrototype>();
+            foreach (_Member d in  await _context.Members.ToArrayAsync())
```

Comment:
Rather assign ` await _context.Members.ToArrayAsync()` to a variable and iterate over the results. That's a cleaner way of doing it and will make debugging a lot easier.

Reply:
**thank you for your views (**:  we'll consider that in later stages

Reply:
No, you won't. The whole point is to avoid having to "consider that in later stages". Projects grow, so there's always more work to be done. If you leave that now then you'll probably never come back to change it. The point is to write clean code from the get-go. Don't fall for the "As long as it works" mentality.